src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/X509CertificateCollection.cs

```diff
@@ -68,33 +52,26 @@ bool IList.IsReadOnly
             get { return false; }
         }
 
-        Object IList.this[int index]
+        object IList.this[int index]
         {
             get
             {
-                if (index < 0 || index >= Count)
-                    throw new ArgumentOutOfRangeException("index", SR.ArgumentOutOfRange_Index);
-
                 return _list[index];
             }
             set
             {
-                if (index < 0 || index >= Count)
-                    throw new ArgumentOutOfRangeException("index", SR.ArgumentOutOfRange_Index);
                 if (value == null)
                     throw new ArgumentNullException("value");
 
-                _list[index] = value;
+                ((IList)_list)[index] = value;
```

Comment:
Why use the IList version? The only difference I see is that the IList version throws on null arguments, but you left the null-check here.

Oh... right, the IList version takes `object`, but the `List<T>` version takes a `T`.

A comment to the effect of "List already has an IList indexer which handles the type and range checks, let it do all the work rather than have it replicated here" would save on some thinking for the next guy.